### DISCUSSION TITLE
Problem with ActivityCallback

### DISCUSSION THREAD

**Discussion by bdyhsj** [ionic-team/capacitor#5994] [Q&A] [answered]
Hi,

We are in the process of developing a capacitor plugin for Android, requiring a proprietary SDK.
One of the functions starts an Activity from which we need the result, and we are able only to dictate which request code to use.

It looks something like this:
```Java
client.x(activity, data, REQUEST_CODE)
```

The `client.x` function then starts an Activity and returns a result with the REQUEST_CODE as in android's own startActivityForResult, but we do not have access to the code doing this.

Do you have any suggestions on how we can setup our plugin to handle this request code, without starting the activity through capacitor's startActivityForResult?

**Comment by jcesarmobile** [comment, marked as answer]
If the SDK is using `startActivityForResult` then you can't use ActivityCallback.

The SDK should be updated to not use `startActivityForResult`, because it's deprecated. But in the meantime you'll have to use the `requestCodes` annotation and the `handleOnActivityResult` method in your plugin as it was done for Capacitor 2.
https://capacitorjs.com/docs/v2/plugins/android#intents-with-results
